env/work/srw_python/srwl_bl.py

```diff
@@ -1695,10 +1695,24 @@ def calc_all(self, _v, _op):
         plotOK = False
 
         if (_v.tr == True) and (len(_v.tr_pl) > 0):
-            #uti_plot1d(int_ss, [mesh_ss.eStart, mesh_ss.eFin, mesh_ss.ne], ['Photon Energy', 'Intensity', 'Intensity'], ['eV', 'ph/s/.1%bw/mm^2'])
-
-            #To implement!
-            #ddddddddddddddddddddddddddddddddddddddddddddddddd
+            args = []
+            kwargs = {
+                'labels': ['Longitudinal Position'],
+                'units': ['m', 'm'],
+            }
+            traj_rep_allowed_values = ['xz', 'yz']
+            traj_rep_allowed_values += [x.upper() for x in traj_rep_allowed_values]
+            if _v.tr_pl.lower() == 'xz':
+                args.append(trj.arX)
+                kwargs['labels'].append('Horizontal Position')
+            elif _v.tr_pl.lower() == 'yz':
+                args.append(trj.arY)
+                kwargs['labels'].append('Vertical Position')
+            else:
```

Comment:
We have to re-iterate on / re-discuss this. Here are my concerns:
1) The trajectory-related plots that we show in SRW are always vs "z" (longitudinal position), so "z" doesn't need to be present in the possible option values (e.g. instead of 'xz' we could have just 'x'). 
2) The trajectory-related set of plots should include not only the horizontal and vertical positions, but also horizontal and vertical angles, and maybe magnetic field components seen by electrons. The corresponding option names could be called 'xp', 'yp', 'bx', 'by'. The corresponding columns are usually present in the trajectory ASCII files generated by SRW (magn. field is not always present though; this can be checked, e.g. based on number of columns in the ASCII file).

Reply:
After re-thinking: let me try to test this and add the other cases (as described above) myself. I will keep the existing allowed values 'xz', 'yz' (so that no changes will be necessary in Sirepo), and will try to add such plots as 'xy' and 'xpyp', which are used sometimes for elliptical undulator characterization.

Reply:
Sounds great! I'll pay attention on other comments meanwhile.

Reply:
OK, done; but I had to re-write it to support generation of multiple trajectory-related plots from one call, e.g. "xz,yz,xpz,ypz,yx,ypxp". I also added yet another list-plot (on irregular mesh) function: uti_plot.uti_plot1d_ir(ary, arx, labels, units) to support such plot cases as "yx,ypxp". Tested, seems to work.

Reply:
Great!